packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -333,6 +334,19 @@ export default class Extension {
       pair.decodePkcs8(password);
     }
 
+    if (Object.prototype.hasOwnProperty.call(request.payload, 'genesisHash')) {
+      // Get the metadata for the genesisHash
+      const currentMetadata = this.#state.knownMetadata.find((meta: MetadataDef) =>
+        meta.genesisHash === (request.payload as SignerPayloadJSON).genesisHash);
+
+      // set the registry before calling the sign function
+      registry.setSignedExtensions((request.payload as SignerPayloadJSON).signedExtensions, currentMetadata?.userExtensions);
```

Comment:
Don't believe his makes sense - the payload will contain all the extensions as per the metadata, so these are known.